proseco/guide.py

```diff
@@ -606,6 +606,25 @@ def check_column_spoilers(cand_stars, ok, stars, n_sigma):
     return column_spoiled, rej
 
 
+def get_ax_range(n, extent):
+    """
+    Given a float pixel row or col value and an integer pixel "extent",
+    return a range for the row or col that is divisible by 2 and contains
+    at least the requested extent
+    :param n: row or col float value
+    :param extent: half of desired range from n
+    :returns: tuple of range as (minus, plus)
+    """
+    minus = int(np.floor(n - extent))
+    plus = int(np.ceil(n + extent))
+    if (np.floor(n) != np.ceil(n)):
```

Comment:
I don't understand this logic.  Basically you want `plus` and `minus` to be even, right?  So would this work?
```
# Ensure plus and minus are both even, expanding the range as needed.
if minus % 2 != 0:
    minus -= 1
if plus % 2 != 0:
    plus += 1
```

Reply:
We want (plus - minus) to be even.  If you ensure that both are even (or odd) you do win that, but you can end up adding extra pixels to both ends that really aren't going to be in the 8x8.

Reply:
With the logic that is in there now, the initial
``` 
minus = int(np.floor(n - extent))
plus = int(np.ceil(n + extent))
```
guarantees you have a odd range (for real float values not exactly at the pixel boundary), and then it adds one more pixel to the side closest to the offset of the star within the center pixel.

Reply:
If this is about figuring out the readout window then can we use the usual logic for defining a readout window?
```
n_int = int(round(n))  # for `n` in edge pixel coords
plus = n_int + extent
minus = n_int - extent
```

Reply:
The extent is a float value.  If you want to round it up, this works too, but will get extra pixels.

Reply:
Where's the usual logic for defining a readout window?  If I should always be taking one of the minus or adding one to the plus anyway, that's fine, but practically, I was trying not to add extra pixels while still making it possible to bin 2x2.  If that's just too ugly we can do something different, but may end up with regression test changes.

Reply:
Here's one implementation of the "usual logic":
  https://github.com/sot/chandra_aca/blob/master/chandra_aca/aca_image.py#L451
I'm sure you find it in annie as well.

I don't quite follow what you are saying, but the 3 lines I wrote will always give the minimal set of pixels that are centered on `n` and can be binned 2x2.  I suspect it is exactly equivalent to what you wrote.

Reply:
I'm still confused.  extent is meant to have 4 pixels for the 8x8 image and, generally, plus 1.6 pix for 8x8 dither.  So plus = n_int + 5.6 doesn't quite work, but if I round up in both directions again that probably works.

So, for a star with nominal row=0.7 in edge coordinates, what should be dithered pixel region be in row?

-6 to 6
or
-5 to 7

Because I thought -5 to 7 was the right answer, but I didn't have the usual logic as my reference.